### ISSUE TITLE
Selectively training a branch of multi-output model

### ISSUE THREAD

**Issue by sahnimanas** [fchollet/keras#6806]
I have a multi output Keras model with a structure resembling this:

```
s = some_shared_layers()(input)
non_trainable1 = Dense(trainable=False) (s) 
non_trainable2 = Dense(trainable=False) (s) 
trainable = Dense() (s) 

model = Model(input, outputs=[non_trainable1, non_trainable2, trainable])
```

My model first computes a forward pass and uses the 1st 2 outputs to manipulate the input. It then computes another forward pass to get the 3rd output.

```
out1, out2,_ =model.predict(input_data) 
processed_data = foo(input_data, out1, out2) 
_,_, out3 = model.predict(processed_data)
```

How should I call `model.fit()` to train only the `trainable` layer? If I exclude the losses for the other outputs, Keras warns `we will not be expecting any data to be passed to "non_trainable1" during training` and excludes them from the computational graph.

Is there a better way to structure the model for this task?

**Comment by bstriner** [comment]
You could hypothetically set the loss and loss_weights to `None` for those outputs. `model.compile('adam', loss=[None,None,whatever], loss_weights=[None,None,1.],...)`. When you train, `model.fit(inputs, [None,None,target])`. That seems a little awkward though and requires a lot of knowledge of internals. Easier to just take advantage of sharing weights.

Make one model for prediction that has 3 outputs. Make one model for training that has one output. They will share weights if you build them correctly, just be careful with saving and loading. Don't even need to worry about `trainable=False` in that situation.

Cheers

**Comment by sahnimanas** [comment]
@bstriner thanks. 2 separate models is what I'm going with for now and it does the job, with a caveat. I wanted to use an ImageDataGenerator on model2 (single output) and thought I could perhaps call model1 inside a `preprocessing_function` passed to the generator. 
However, running `model2.fit_generator` again led to the same issue (unshared layers not part of the graph)  

For now, I'm running the original input on model1 separately, dumping outputs to a .npy file, and then passing that to the generator when I later train model2.

Let me know if there's a workaround to this
